nilearn/datasets/func.py

```diff
@@ -784,33 +781,32 @@ def fetch_localizer_contrasts(
     root_url = "https://osf.io/download/{0}/"
     files = {}
     filenames = []
-    for subject_id in subject_ids:
-        for data_type in data_types:
-            for contrast in contrasts_wrapped:
-                name_aux = f"{data_type}_{contrast}"
-                name_aux.replace(" ", "_")
-                file_path = Path(
-                    "brainomics_data", subject_id, f"{name_aux}.nii.gz"
-                )
 
-                path = "/".join(
-                    [
-                        "/localizer",
-                        "derivatives",
-                        "spm_1st_level",
-                        f"sub-{subject_id}",
-                        (
-                            f"sub-{subject_id}_task-localizer"
-                            f"_acq-{contrast}_{data_type}.nii.gz"
-                        ),
-                    ]
-                )
+    for subject_id, data_type, contrast in itertools.product(
+        subject_ids, data_types, contrasts_wrapped
+    ):
+        name_aux = f"{data_type}_{contrast}"
+        name_aux.replace(" ", "_")
+        file_path = Path("brainomics_data", subject_id, f"{name_aux}.nii.gz")
 
-                if _is_valid_path(path, index, verbose=verbose):
-                    file_url = root_url.format(index[path][1:])
-                    opts = {"move": file_path}
-                    filenames.append((file_path, file_url, opts))
-                    files.setdefault(data_type, []).append(file_path)
+        path = "/".join(
+            [
+                "/localizer",
+                "derivatives",
+                "spm_1st_level",
+                f"sub-{subject_id}",
+                (
+                    f"sub-{subject_id}_task-localizer"
+                    f"_acq-{contrast}_{data_type}.nii.gz"
+                ),
+            ]
+        )
```

Comment:
I am wondering if this can also be refactored to a Path object?

Reply:
I think I tried this in the past and there was a good reason not to this, but I cannot remember.

In any case I would say this is beyond the scope of the PR.

Reply:
but feel free to try it in another PR. 😜

Reply:
> I think I tried this in the past and there was a good reason not to this, but I cannot remember.

Looked at it again and it looks like we are trying to make a URL not a path so Path might give weird URL on windows machine